docs/how_to/compose_quantum_circuits.rst

```diff
@@ -0,0 +1,180 @@
+########################
+Compose quantum circuits
+########################
+
```

Comment:
I didn't know you could compose with `+` in the past. I can add a warning about that.

Reply:
FYI. Its part of upgrade notes here https://qiskit.org/documentation/release_notes.html#release-notes-0-23-0-upgrade-notes

>The [QuantumCircuit](https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.html#qiskit.circuit.QuantumCircuit) methods combine and extend have been removed following their deprecation in Qiskit Terra 0.17.0. This was done because these functions were simply less powerful versions of [QuantumCircuit.compose()](https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.compose.html#qiskit.circuit.QuantumCircuit.compose), which should be used instead.
>  The removal of extend also means that the + and += operators are no longer defined for [QuantumCircuit](https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.html#qiskit.circuit.QuantumCircuit). Instead, you can use the & and &= operators respectively, which use [QuantumCircuit.compose()](https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.compose.html#qiskit.circuit.QuantumCircuit.compose).